### PR TITLE
Develop -> Master

### PR DESCRIPTION
**Pull request by bio-boris**

### CODE REVIEW

File: src/us/kbase/common/utils/CondorUtils.java [hunk 1]
```diff
@@ -128,11 +130,11 @@ public static CondorResponse runProcess(String[] condorCommand) throws IOExcepti
             e.printStackTrace();
         }
 
-        if(process.isAlive()){
+        CondorResponse failure = new CondorResponse(stdOutMessage, stdErrMessage, false);
+        if (process.isAlive()) {
```
Comment (slebras) [inline src/us/kbase/common/utils/CondorUtils.java]: we want this to fail if isn't completed in 30 seconds?

File: src/us/kbase/common/utils/CondorUtils.java [hunk 2]
```diff
@@ -239,15 +240,20 @@ public static String submitToCondorCLI(String ujsJobId, AuthToken token, String
...
 
+        String stderr = null;
         while (jobID == null && retries > 0) {
-            jobID = runProcess(cmdScript).stdout.get(0);
+            CondorResponse r = runProcess(cmdScript);
+            if (r.success)
+                jobID = r.stdout.get(0);
+
+            stderr = String.join(", ", r.stderr);
             retries--;
```
Comment (slebras) [inline src/us/kbase/common/utils/CondorUtils.java]: would it make sense to add some time delay between attempts here?

File: src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java [hunk 3]
```diff
@@ -1,240 +1,268 @@
...
+			final Map<String, Object> ret = new HashMap<>();
+			for (final String k: m.keySet()) {
+				if (!k.equals("_id")) {
+					final Object v = m.get(k);
+					ret.put(k, cleanObject(v));
+				}
+			}
+			return ret;
+		}
+		return dbo;
```
Comment (slebras) [inline src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java]: is this an error case if it the input `dbo` isnt one of the spcified types?

File: src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java [hunk 4]
```diff
@@ -1,240 +1,268 @@
...
 	}
 
 	public void updateExecLogLines(String ujsJobId, int newLineCount,
 								   List<ExecLogLine> newLines) throws Exception {
-		execLogs.update(String.format("{%s:#}", PK_EXEC_LOGS), ujsJobId).with(
-				String.format("{$set:{%s:#,%s:#},$push:{%s:{$each:#}}}",
-						"original_line_count", "stored_line_count", "lines"),
-				newLineCount, newLineCount, newLines);
+		// needs input checking
+		logCol.update(new BasicDBObject(PK_EXEC_LOGS, ujsJobId),
```
Comment (slebras) [inline src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java]: this section is a little bit of a nightmare to look at. Can it be split up for readability?

File: src/us/kbase/narrativejobservice/sdkjobs/SDKLocalMethodRunner.java [hunk 5]
```diff
@@ -522,10 +522,10 @@ public boolean isJobCanceled() {
                             return true;
                         }
                     } catch (Exception ex) {
-                        log.logNextLine("Non-critical error checking for job cancelation - " +
+                        log.logNextLine("ineffective attempt checking for job cancelation - " +
```
Comment (slebras) [inline src/us/kbase/narrativejobservice/sdkjobs/SDKLocalMethodRunner.java]: how about "status check failed". "ineffective attempt" is definitely better than "Non-critical error" though

File: src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java [hunk 6]
```diff
@@ -153,9 +153,13 @@ public static String runJob(RunJobParams params, AuthToken authPart,
 			String newExternalURL = config.get(NarrativeJobServiceServer.CFG_PROP_SELF_EXTERNAL_URL);
 			String parentJobId = params.getParentJobId();
 			String schedulerType = "condor";
-
-			String condorId = CondorUtils.submitToCondorCLI(ujsJobId, authPart, aweClientGroups, newExternalURL, baseDir, optClassAds, getCatalogAdminAuth(config));
-			saveTask(ujsJobId, condorId, jobInput, appJobId, schedulerType,parentJobId, config);
+			try {
+				String condorId = CondorUtils.submitToCondorCLI(ujsJobId, authPart, aweClientGroups, newExternalURL, baseDir, optClassAds, getCatalogAdminAuth(config));
+                saveTask(ujsJobId, condorId, jobInput, appJobId, schedulerType,parentJobId, config);
```
Comment (slebras) [inline src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java]: weird spacing here

File: src/us/kbase/common/utils/CondorUtils.java (hunk 1 above)
Comment (bio-boris) [reply src/us/kbase/common/utils/CondorUtils.java]: yes

File: src/us/kbase/common/utils/CondorUtils.java (hunk 2 above)
Comment (bio-boris) [reply src/us/kbase/common/utils/CondorUtils.java]: There is already a time delay in running condor_q already of 1-3 seconds. I would have to decrease the number of retries if I put a delay in.

File: src/us/kbase/narrativejobservice/sdkjobs/SDKLocalMethodRunner.java (hunk 5 above)
Comment (bio-boris) [reply src/us/kbase/narrativejobservice/sdkjobs/SDKLocalMethodRunner.java]: The goal is to try to come up with a less scary way of saying this. `Failed` = scary

File: src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java (hunk 6 above)
Comment (bio-boris) [reply src/us/kbase/narrativejobservice/sdkjobs/SDKMethodRunner.java]: fixed

File: src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java (hunk 4 above)
Comment (bio-boris) [reply src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java]: Is
```
		BasicDBObject ujsJobIdQuery = new BasicDBObject(PK_EXEC_LOGS, ujsJobId);

		BasicDBObject each = new BasicDBObject("lines", new BasicDBObject("$each",
				newLines.stream().map(l -> toDBObj(l))
						.collect(Collectors.toList())));

		BasicDBObject set = new BasicDBObject("$set", new BasicDBObject("original_line_count", newLineCount).append("stored_line_count", newLineCount))
				.append("$push", each);

		logCol.update(ujsJobIdQuery,set);
```
Better or worse?

Comment (slebras) [reply src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java]: For me I think it's better.  Maybe @MrCreosote would have an opinion on it as well.

File: src/us/kbase/narrativejobservice/sdkjobs/SDKLocalMethodRunner.java (hunk 5 above)
Comment (slebras) [reply src/us/kbase/narrativejobservice/sdkjobs/SDKLocalMethodRunner.java]: gotcha, I think this is good for now

File: src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java (hunk 4 above)
Comment (MrCreosote) [reply src/us/kbase/narrativejobservice/db/ExecEngineMongoDb.java]: lol no

**Pull request merged**